#25 implement full archive for chat

amiedavis opened the pull request:
Delivers #25

amiedavis committed:
#14 Fixed constant name in SyllabusArchiver and used it in archiveContent() calls

amiedavis committed:
#25 Added method to Jsonifier to deal with dates.

amiedavis committed:
#25 Added archiver implementation for chat - WIP

Need to investigate error when using Jsonifier.

amiedavis committed:
#25 Removed date method from Jsonifier

Didn't end up needing this extra method.

amiedavis committed:
#25 Finished implementation of archiver for chat

amiedavis committed:
Merge branch 'master' into #25-implement-full-archive-for-chat

# Conflicts:
#	pom.xml

amiedavis committed:
#25 Moved the creation of the archive item to a separate function.

steveswinsburg commented on provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java:
```diff
@@ -0,0 +1,157 @@
+		for (ChatChannel chatChannel : chatChannels) {
+
+			// Save the metadata for this channel
+			ChatChannelMetadata metadata = new ChatChannelMetadata();
```
This block could go to new method like the others

amiedavis replied:
durp

steveswinsburg commented on provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java:
```diff
@@ -0,0 +1,157 @@
+	/**
+	 * Simplified helper class to represent an individual chat message
+	 */
+	private static class SimpleChatMessage {
```
-static

steveswinsburg commented on provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java:
```diff
@@ -0,0 +1,157 @@
+	/**
+	 * Simplified helper class to represent the metadata for a chat channel
+	 */
+	private static class ChatChannelMetadata {
```
-static

steveswinsburg commented on provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java:
```diff
@@ -0,0 +1,157 @@
+			simpleChatMessage.setBody(message.getBody());
+			simpleChatMessage.setDate(message.getMessageDate());
+			try {
+				simpleChatMessage.setOwner(userDirectoryService.getUser(message.getOwner()).getDisplayName());
```
These two calls can be condensed to one. And use a method to get the user. If user doesn't exist, use the uuid. Also userId in simple chat message should be eid.

steveswinsburg replied:
If you have a list, call it something plural. Then inside you make items.those can be a method on their own that return just one thing.

So createArchiveItems and createArchiveItem

amiedavis replied:
Which two calls can be condensed into one? I'm confused.
Also aren't I already using the eid?

amiedavis committed:
#25 Fixes as per PR comments

Set metadata content in separate own method
Set message content in separate method
Get user in separate method
If user doesn't exist, use uuid in place of display name and eid
Remove "static" from helper object inner classes

amiedavis committed:
#25 Fixed numbers in filenames

amiedavis committed:
Merge branch 'master' into #25-implement-full-archive-for-chat

# Conflicts:
#	pom.xml
#	provider-syllabus/src/java/org/sakaiproject/archiver/provider/SyllabusArchiver.java
